site/public/js/calendar.js

```diff
@@ -772,3 +772,21 @@ function colorLegend() {
     });
 }
 
+//args contains course id and course name. We want to use display_name is possible
+function filter_course(args) {
```

Comment:
```suggestion
function filter_course(course_val, display_name) {
```

Following my previous suggestion of taking in two arguments rather than a string.